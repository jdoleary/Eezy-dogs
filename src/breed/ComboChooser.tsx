import React from 'react';
import { Grid, Box, TextField, IconButton } from '@material-ui/core';
import AddCircle from '@material-ui/icons/AddCircle';
import RemoveCircle from '@material-ui/icons/RemoveCircle';
import NotInterested from '@material-ui/icons/NotInterested';

import { addCombo, removeCombo, changeCombo, Combo, store } from '../store';
import Selector from './Selector';
import type { BreedMap } from './actions';

interface ComboChooserProps {
  combo: Combo;
  lastInList: boolean;
  index: number;
  breeds: BreedMap;
}
export default function ComboChooser({
  combo,
  lastInList,
  index,
  breeds,
}: ComboChooserProps) {
  const { breed, subBreed, count } = combo;
  const isFirstAndOnly = index === 0 && lastInList;
  return (
    <Box m={2}>
      <Grid container justify="flex-start">
        <Grid item>
          <Selector
            name="Breed"
            options={Object.keys(breeds)}
            value={breed}
            onChange={(_event: any, breed: string) => {
              store.dispatch(
                changeCombo({ combo: { ...combo, breed }, index }),
              );
            }}
          />
        </Grid>
        <Grid item>
          <Selector
            name="SubBreed"
            options={breeds[breed]}
            value={subBreed}
            onChange={(_event: any, subBreed: string) => {
              store.dispatch(
                changeCombo({ combo: { ...combo, subBreed }, index }),
              );
            }}
          />
        </Grid>
        <Grid item>
          <Box width={'100px'}>
            <TextField
              variant="outlined"
              type="number"
              value={count}
              onChange={(event: any) => {
                store.dispatch(
                  changeCombo({
                    combo: { ...combo, count: event.target.value },
                    index,
                  }),
                );
              }}
            ></TextField>
          </Box>
        </Grid>
        <Grid item>
          <Box component="div" display="flex" flexWrap="nowrap">
            <Box component="div">
              <IconButton
                color="primary"
                aria-label={isFirstAndOnly ? 'clear row' : 'remove row'}
                component="span"
                onClick={() => {
                  // Remove the current combo
                  store.dispatch(removeCombo(index));
                  if (isFirstAndOnly) {
                    // If removing the first and only combo, replace it with a new, empty one
                    store.dispatch(
                      addCombo({ breed: '', subBreed: '', count: 1 }),
                    );
                  }
                }}
              >
                {isFirstAndOnly ? <NotInterested /> : <RemoveCircle />}
              </IconButton>
            </Box>
            <Box component="div" visibility={lastInList ? 'visible' : 'hidden'}>
              <IconButton
                color="primary"
                aria-label="add row"
                component="span"
                onClick={() => {
                  store.dispatch(
                    addCombo({ breed: '', subBreed: '', count: 1 }),
                  );
                }}
              >
                <AddCircle />
              </IconButton>
            </Box>
          </Box>
        </Grid>
      </Grid>
    </Box>
  );
}
