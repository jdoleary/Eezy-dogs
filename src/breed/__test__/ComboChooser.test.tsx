import React from 'react';
import { describe, expect, it } from '@jest/globals';
import renderer from 'react-test-renderer';
import ComboChooser from '../ComboChooser';

describe('ComboChooser', () => {
  it('Should match snapshot', () => {
    const component = renderer.create(
      <ComboChooser
        combo={{ breed: 'breed', subBreed: 'subBreed', count: 23 }}
        lastInList={false}
        index={0}
        breeds={{
          breed: ['subBreed'],
        }}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
