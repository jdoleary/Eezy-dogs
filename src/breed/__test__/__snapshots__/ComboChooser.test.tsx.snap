// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboChooser Should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="breed"
            id="breed-label"
          >
            Breed
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              disabled={false}
              id="breed"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={false}
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Breed
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="subbreed"
            id="subbreed-label"
          >
            SubBreed
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              disabled={false}
              id="subbreed"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={false}
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  SubBreed
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiBox-root MuiBox-root-6"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value={23}
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-3"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiBox-root MuiBox-root-7"
      >
        <div
          className="MuiBox-root MuiBox-root-8"
        >
          <span
            aria-disabled={false}
            aria-label="remove row"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          className="MuiBox-root MuiBox-root-9"
        >
          <span
            aria-disabled={false}
            aria-label="add row"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
